tests/components/template/test_cover.py

```diff
@@ -370,26 +494,19 @@ async def test_template_state_text_with_position(
     [
         (ConfigurationStyle.LEGACY, "position_template"),
         (ConfigurationStyle.MODERN, "position"),
+        (ConfigurationStyle.TRIGGER, "position"),
     ],
 )
-@pytest.mark.parametrize(
-    "set_state",
-    [
-        "",
-        None,
-    ],
-)
```

Comment:
Why is this removed?

Reply:
I disagree with the behavior.  `""` is not a valid state and it _should_ produce an error in the logs.  I can add it back but I'll likely have to separate out Trigger because I'd have to override the process data method in order to allow empty strings without error in trigger based entities.

What are your thoughts?

Reply:
OK, but the behavior does not change just because you remove the test parametrization? Or do you mean the test is not valid?

Reply:
This is the same test, also added by @jbouwh in PR https://github.com/home-assistant/core/pull/91172

Reply:
When MQTT templates are used, an empy result indicates there is no state change. MQTT state topics are reused, and the templates filter out the states. This is a common pattern for mqtt platforms.

![afbeelding](https://github.com/user-attachments/assets/b07495fb-8cff-44dd-9bcc-4966b677d68a)